benchmarks/mathvista/mathvista.py

```diff
@@ -0,0 +1,160 @@
+
+{question}
+
+{choices}
+""".strip()
+
+FREEFORM_TEMPLATE = r"""
+Answer the following question. The entire content of your response should be of the following format: 'ANSWER: $ANSWER' (without quotes) where $ANSWER is one your answer.
+
+{question}
+"""
+
+
+@task
+def MathVistaMultipleChoice() -> Task:
+    dataset = hf_dataset(
+        path="AI4Math/MathVista",
+        split="testmini",
+        sample_fields=record_to_sample,
+        trust=True,
+        shuffle=True,
+    )
+
+    return Task(
+        dataset=dataset,
+        plan=[multi_choice_or_free_form_solver()],
+        scorer=multi_choice_or_free_form_scorer(),
+    )
+
+
+def get_multi_choice_as_letter(record: dict) -> chr:
+    choices = record["choices"]
+    labels = [chr(i) for i in range(ord("a"), ord("a") + len(choices))]
+
+    choices = dict(zip(labels, choices))
+
+    answer = record["answer"]
+    target = list(choices.values()).index(answer)
+    target = chr(ord("A") + int(target))
+
+    return target
+
+
+def record_to_sample(record: dict) -> Sample:
+    message = [
+        ChatMessageUser(
+            content=[
+                ContentText(text=record["question"])
+                if record["question_type"] == "multi_choice"
+                else ContentText(text=record["query"]),
+                ContentImage(image=record["image"]),
+            ]
+        )
+    ]
+    if record["question_type"] == "multi_choice":
+        return Sample(
+            input=message,
+            choices=record["choices"],
+            target=get_multi_choice_as_letter(record),
+            id=record["pid"],
+            metadata={
+                "question_type": record["question_type"],
+                "answer_type": record["answer_type"],
+                **record["metadata"],
+            },
+            files={f"image:{record['image']}": record["image"]},
+        )
+
+    elif record["question_type"] == "free_form":
+        return Sample(
+            input=message,
+            target=record["answer"],
+            id=record["pid"],
+            metadata={
+                "precision": record["precision"],
+                "question_type": record["question_type"],
+                "answer_type": record["answer_type"],
+                **record["metadata"],
+            },
+            files={f"image:{record['image']}": record["image"]},
+        )
+
+
+@solver
+def multi_choice_or_free_form_solver() -> Solver:
+    async def solve(state: TaskState, generate: Generate) -> TaskState:
+        if state.metadata["question_type"] == "multi_choice":
+            state.user_prompt.text = prompt(
+                question=state.user_prompt.text,
+                choices=state.choices,
+                template=SINGLE_ANSWER_TEMPLATE,
+            )
+
+            return await generate(state)
+
+        elif state.metadata["question_type"] == "free_form":
```

Comment:
Probably should throw if we make it past the `elif` without matching "multi_choice" or "free_form"